angr/engines/pcode/lifter.py

```diff
@@ -427,6 +427,13 @@ def constant_jump_targets_and_jumpkinds(self):
 
         return exits
 
+    @property
+    def is_noop_block(self) -> bool:
+        """
+        Returns True if this block is a no-op block (i.e. it has no instructions and no jumps).
+        """
+        return len(self._ops) == 0
```

Comment:
I believe this method should return `True` when the block contains instructions that don't have any effects (e.g. just IMark ops). ex:

```
import pypcode
ctx = pypcode.Context("x86:LE:64:default")
tx = ctx.translate(b"\x90\x90")
for op in tx.ops:
    print(pypcode.PcodePrettyPrinter.fmt_op(op))

IMARK ram[0:1]
IMARK ram[1:1]
```

So probably filter with:

```py
any(op.opcode != pypcode.OpCode.IMARK for op in tx.ops)
```